airflow-core/src/airflow/jobs/triggerer_job_runner.py

```diff
@@ -870,8 +870,16 @@ async def create_triggers(self):
             await asyncio.sleep(0)
 
             try:
-                kwargs = Trigger._decrypt_kwargs(workload.encrypted_kwargs)
-                trigger_instance = trigger_class(**kwargs)
+                from airflow.serialization.serialized_objects import smart_decode_trigger_kwargs
+
+                # Decrypt and clean trigger kwargs before for execution
+                # Note: We only clean up serialization artifacts (__var, __type keys) here,
+                # not in `_decrypt_kwargs` because it is used during hash comparison in
+                # add_asset_trigger_references and could lead to adverse effects like hash mismatches
+                # that could cause None values in collections.
+                kw = Trigger._decrypt_kwargs(workload.encrypted_kwargs)
+                deserialised_kwargs = {k: smart_decode_trigger_kwargs(v) for k, v in kw.items()}
```

Comment:
Can this not just use `BaseSerialization.deserialize`?

Reply:
We should probably fix the docstring of `smart_decode_trigger_kwargs` in this case since it’s no longer just used for UI.

Reply:
Oh I saw you already discussed that below.

Reply:
Yeah :)